src/Microsoft.DotNet.Darc/DarcLib/IGitRepo.cs

```diff
@@ -26,4 +27,17 @@ public interface IGitRepo
     /// <param name="branch">Branch to push to</param>
     /// <param name="commitMessage">Commit message</param>
     Task CommitFilesAsync(List<GitFile> filesToCommit, string repoUri, string branch, string commitMessage);
+
+    async Task<bool> IsRepoVmrAsync(string repoUri, string branch)
+    {
+        try
+        {
+            await GetFileContentsAsync(VmrInfo.DefaultRelativeSourceMappingsPath, repoUri, branch);
+            return true;
+        }
+        catch
```

Comment:
Technically it would be best if we only caught the 404s - if a connection fails, it will mistakenly think a repo is not a VMR and wreak havoc

Reply:
a 404 won't be enough because this also works for local repos, will look into what fits best here

Reply:
I'll move the implementation out of here and put in the classes so they can catch what they need to

Reply:
Each implementation should probably catch whatever native to them and re-throw a unified type

Reply:
yup, doing that now

Reply:
Ah looks like we do have an exception we throw in all implementations `DependencyFileNotFoundException` will use that